merlin/systems/dag/ops/faiss.py

```diff
@@ -131,39 +142,48 @@ def export(
         self_params.update(params)
         index_filename = os.path.basename(os.path.realpath(self.index_path))
 
-        # set index path to new path after export
-        full_path = pathlib.Path(path) / f"{node_id}_{QueryFaiss.__name__.lower()}" / str(version)
+        if backend == "ensemble":
```

Comment:
Why does this conditional export only apply to the Faiss op and not all the others?

Reply:
`QueryFAISS` is the only Python op currently saving additional files to disk (aside from the Triton config and `model.py`.) In an `ensemble` export, we write the FAISS index file into the `n_queryfaiss` directory in the model repository, but that directory doesn't exist with an `executor` export, so instead we write it to the `executor_model` directory. If other ops were also writing additional files, this conditional would apply there too.

(But should disappear when we introduce backends/runtimes/adapters, since then we can have separate `export` methods for each export format.)

Reply:
The Implicit op also writes out files to it's directory. Is that a different kind of file in this context?

Reply:
Ah, good point! That should probably write its files to the `executor_model` directory too then.